[BC-BREAKING] Add scalar type info to tensor print

li-roy opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#20483 Add scalar type info to tensor print**

This got taken out when Type became scalar type agnostic. Adding it back. closes #20320

Differential Revision: [D15334010](https://our.internmc.facebook.com/intern/diff/D15334010)

Commit:
Add scalar type info to tensor print

ezyang reviewed (approved):
Technically, this is BC-breaking as you changed the signature of `toString`, but it's probably benign.

Commit:
Update on "[BC-BREAKING] Add scalar type info to tensor print"

Add scalar type info to tensor print

gh-metadata: pytorch pytorch 20483 gh/li-roy/19/head
